src/requirements.ts

```diff
@@ -71,7 +71,7 @@ function checkJavaRuntime(): Promise<string> {
 
 function readJavaConfig(): string {
     const config = workspace.getConfiguration();
-    return config.get<string>('java.home', null);
+    return config.get<string>('java.home', '');
```

Comment:
You removed the config contribution so I don't think this will work. Has it been tested without JAVA_HOME being set ?

Reply:
If you don't declare the config property in package.json, then I don't think the API will work and it's likely that you will get an exception (from my vscode knowledge)

Reply:
In this case if the property java.home is not found, the default value '' (empty string) is returned, which will be evaluated false in the caller method and the code executed will be the else part as it was before in the old server connector version.

Reply:
This looks fine to me. I could revert it to null but Lucca seems correct so it's not worth wasting a commit.